bridge/ffi/src/lib.rs

```diff
@@ -9,12 +9,42 @@
 //! To use the FFI layer, you need to build the dynamic library (e.g., .so, .dylib, or .dll) by adding
 //! the appropriate configuration to `Cargo.toml`. Then, include the generated dynamic library and
 //! the C header file `dsnp_graph_sdk_ffi.h` in your C, C++, or other FFI-compatible projects.
+
 mod c_api;
 pub use c_api::*;
 mod bindings;
 pub use bindings::*;
 mod utils;
 pub use utils::*;
+mod errors;
+pub use errors::*;
 
 #[cfg(test)]
 mod tests;
+
+use std::ptr::NonNull;
+
+#[repr(C)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct DsnpGraphFFIResult<T, E> {
+	pub result: Option<NonNull<T>>,
+	pub error: Option<NonNull<E>>,
```

Comment:
I was imagining that maybe C enum Error might be sufficient. Any reason why we need generic (dynamic) errors?

Reply:
Just a design pattern so any changes to error types wont warrant changing this generic type always, but yes can be pinned to exact error type too, either or works